### ISSUE TITLE
Mandatory to use one of (as_physics_system, as_visual_system) for each system?

### ISSUE THREAD

**Issue by YPares** [jrockett6/bevy_godot4#5]
I've been checking a bit the implementation, and am I right in assuming that any system added to the bevy app **needs** to use either `as_physics_system` or `as_visual_system`?

Because if a system uses neither, it seems it will run BOTH in the _process and the _physics_process, and therefore that it'll be updated too often. How do the default bevy systems (added from the standard bevy plugins you are using) behave in that context?

If so, would there be a way to defend against that, for instance by running systems by default only in the _process, and moving them to _physics_process if they are explicitly tagged with `as_physics_system`?

**Comment by jrockett6** [comment]
Good call, setting things to use process updates by default and only use physics frames when tagged explicitly would be better.

Likely some of the standard bevy plugins behavior is not as expected - like the FrameCountPlugin and potentially others.

**Comment by YPares** [comment]
Maybe a possibility is to use World and Schedules directly, and discard the App? See for instance the example in https://docs.rs/bevy/latest/bevy/ecs/system/struct.Deferred.html

This may also lighten the build as I believe you'd need only `bevy_ecs` ( https://github.com/bevyengine/bevy/tree/main/crates/bevy_ecs ) and not the whole `bevy` crate.

**Comment by SFilinsky** [comment]
On topic of physics frames vs visual frames: 

I worked a lot [on my fork](https://github.com/SFilinsky/bevy-godot-bridge) in the past half-year. Once I moved to more complex setup with Rapier and synchronization of hundreds of entities and events, system where .update() is called with 2 different system sets started breaking because it conflicts with Bevy internals. 

The problem is that Bevy keeps some events _(`RemovedComponents<>` for example)_ for one frame, and so if it happens that system that needs it runs in different update set it never knows about it. It caused random breaks in gameplay logic and especially sync between Bevy and Godot that were hard to identify.

The solution I went for is updating both `physics` and `visual` part in `process`. To correctly reproduce determenistic physics, we just run full Bevy updates in `BevyApp::process` and give Bevy `min(godot_simulation_time_delta, 50ms)` to avoid exponential performance decline on large stalls. Then all Bevy systems can rely on regular Bevy schedules and use Fixed if needed.

To provide any better solution, I think Bevy and Godot need lower-level integration and `Node::process` is generally not the best place for it. If Bevy update takes too long if freezes Godot rendeing. To work correctly it rather needs separate thread which won't block Godot physics and rendering at all.
